src/aws/flb_s3_local_buffer.c

```diff
@@ -414,3 +414,106 @@ int flb_remove_chunk_files(struct flb_local_chunk *c)
     }
     return ret;
 }
+
+/* Generic replace function for strings. */
+flb_sds_t replace_uri_tokens(const char* original_string, const char* current_word, const char* new_word) 
+{ 
+    flb_sds_t result; 
+    int i = 0;
+    int count = 0; 
+    int new_word_len = strlen(new_word); 
+    int old_word_len = strlen(current_word); 
+  
+    for (i = 0; original_string[i] != '\0'; i++) { 
+        if (strstr(&original_string[i], current_word) == &original_string[i]) { 
+            count++; 
+            i += old_word_len - 1; 
+        } 
+    }
+
+    result = flb_sds_create_size(i + count * (new_word_len - old_word_len) + 1);
+    if (!result) {
+        flb_errno();
+        flb_sds_destroy(result);
```

Comment:
If result is Null, you shouldn't destroy it. In C, you can call free(NULL) safely, but flb_sds_destroy(NULL) will lead to a segfault. Because SDS strings use some pointer magic: https://github.com/antirez/sds#how-sds-strings-work

Reply:
Thanks for pointing this out. I did not know about this.